base/loading.jl

```diff
@@ -1392,6 +1392,13 @@ function run_extension_callbacks(pkgid::PkgId)
     extids = pop!(EXT_DORMITORY, pkgid, nothing)
     extids === nothing && return
     for extid in extids
+        if in(extid.id, precompilation_stack) && !isprecompiled(extid.id)
```

Comment:
I wonder if this `isprecompiled` call will be called a lot. Maybe this should be moved to where the `require` call is that actually loads the extension?